app/services/katello/pulp3/repository.rb

```diff
@@ -268,6 +268,20 @@ def self.orphan_repository_versions(smart_proxy)
         version_hrefs - ::Katello::Repository.where(version_href: version_hrefs).pluck(:version_href)
       end
 
+      def self.orphan_repository_versions_for_mirror(smart_proxy)
+        current_pulp_repositories = ::Katello::Pulp3::Repository.list(smart_proxy, {})
+        repo_hrefs = current_pulp_repositories.collect { |repo| repo[:_href] }.uniq
+
+        version_hrefs = repo_hrefs.collect do |href|
+          repo_versions = ::Katello::Pulp3::Repository.versions_list(
+            smart_proxy, href, ordering: :_created)
+          repo_versions.pop
+          repo_versions.pluck(:_href)
```

Comment:
Just one note here, perhaps switch to `collect` here like you did on line 273 for consistency or change that `collect` to `pluck`